### PR TITLE
Apply method does not fill called function _args variable

### PR DESCRIPTION
**Pull request by parro-it**
While changing [gravity-func](https://github.com/parro-it/gravity-func) to introduce the new apply method, I found what I think could be a bug in `apply` method.

`apply` correctly pass arguments to the called functions, but the `_args` dynamic argument only contains the first argument.

In this PR I add a failing test that check _args variable, and some log. 

The PR is not meant to be merged...

### CODE REVIEW

File: src/runtime/gravity_core.c [hunk 1]
```diff
@@ -1038,7 +1038,11 @@ static bool closure_apply (gravity_vm *vm, gravity_value_t *args, uint16_t nargs
 	gravity_value_t self_value = GET_VALUE(1);
 	gravity_list_t *list = VALUE_AS_LIST(GET_VALUE(2));
 	
-	gravity_vm_runclosure(vm, closure, self_value, list->array.p, (uint16_t)marray_size(list->array));
+	uint16_t argsCount = (uint16_t)marray_size(list->array);
+
+	printf("apply args size: %d\n", argsCount);
+	printf("args: %ld %ld\n", list->array.p[0].n, list->array.p[1].n);
+	gravity_vm_runclosure(vm, closure, self_value, list->array.p, argsCount + 1);
```
Comment (parro-it) [inline src/runtime/gravity_core.c]: Running the new test, these lines output

```
apply args size: 2
args: 1 2
```
That seems correct...

Comment (parro-it) [inline src/runtime/gravity_core.c]: Here, by adding +1 to argsCount, the test pass, all arguments are copied in _args variable.
But seems incorrect and fragile to add 1...

Comment (parro-it) [inline src/runtime/gravity_core.c]: If I don't add it, `_args[0] == 1` and `_args.count == 1`

File: src/runtime/gravity_core.c [hunk 2]
```diff
@@ -1038,7 +1038,11 @@ static bool closure_apply (gravity_vm *vm, gravity_value_t *args, uint16_t nargs
 	gravity_value_t self_value = GET_VALUE(1);
 	gravity_list_t *list = VALUE_AS_LIST(GET_VALUE(2));
 	
-	gravity_vm_runclosure(vm, closure, self_value, list->array.p, (uint16_t)marray_size(list->array));
+	uint16_t argsCount = (uint16_t)marray_size(list->array);
+
+	printf("apply args size: %d\n", argsCount);
+	printf("args: %ld %ld\n", list->array.p[0].n, list->array.p[1].n);
```
Comment (InusualZ) [inline src/runtime/gravity_core.c]: I don't think these printf should end up in the final product

Comment (parro-it) [inline src/runtime/gravity_core.c]: This is not meant to be merged but to showcase a bug... 😀

**Comment by marcobambini** [comment]
Thanks a lot @parro-it, fixed by https://github.com/marcobambini/gravity/commit/4149aeece34638c7030073ebe684cecd96bf04c5 and unit test added.

**Comment by parro-it** [comment]
👏👏

**Pull request closed without merging**